libs/pq_proto/src/lib.rs

```diff
@@ -458,7 +504,7 @@ pub enum BeMessage<'a> {
     CloseComplete,
     // None means column is NULL
     DataRow(&'a [Option<&'a [u8]>]),
-    ErrorResponse(&'a str),
+    ErrorResponse(&'a str, Option<io::ErrorKind>),
```

Comment:
I think mapping from the ErrorKind to the Postgres error code belongs elsewhere.

Suggest having the postgres error code here, and have a separate function to get the appropriate error code from ErrorKind.

```suggestion
    // error message, Postgres error code (SQLSTATE)
    ErrorResponse(&'a str, [u8; 5]),
```

Reply:
Used as part of https://github.com/neondatabase/neon/pull/3227#discussion_r1059434077